### PR TITLE
Convert to v2-addon

### PR DESCRIPTION
**Pull request by dbendaou**
Converting to v2 addon leveraging the [ember-addon-migrator](https://github.com/NullVoxPopuli/ember-addon-migrator) codemod

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,24 +1,25 @@
...
+/.pnp*
+/.pnpm-debug.log
 /.sass-cache
+.eslintcache
 /connect.lock
 /coverage/
 /libpeerconnection.log
 /npm-debug.log*
 /testem.log
 /yarn-error.log
```
Comment (dbendaou) [inline .gitignore:20]: Keep `.DS_store`

File: CHANGELOG.md
Comment (dbendaou) [inline CHANGELOG.md:1]: Need to backport CHANGELOG

File: ember-amount-input/package.json [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    "lint:js": "eslint . --cache",
+    "lint:hbs:fix": "ember-template-lint . --fix --no-error-on-unmatched-pattern",
+    "lint:js:fix": "eslint . --fix",
+    "start": "rollup --config --watch",
+    "test": "echo 'A v2 addon does not have tests, run tests in test-app'",
+    "prepack": "yarn run build",
+    "prepare": "yarn run build"
+  },
+  "dependencies": {
+    "@embroider/addon-shim": "^1.0.0"
```
Comment (dbendaou) [inline ember-amount-input/package.json:32]: Should we keep `ember-cli-htmlbars` & `ember-cli-babel`? 

For babel I'm pretty sure no but just checking

File: package.json [hunk 3]
```diff
@@ -1,90 +1,32 @@
 {
   "name": "ember-amount-input",
-  "version": "5.0.0",
-  "description": "The default blueprint for ember-cli addons.",
-  "keywords": [
-    "ember-addon"
-  ],
-  "homepage": "https://qonto.github.io/ember-amount-input/versions/master",
+  "version": "0.0.0",
```
Comment (dbendaou) [inline package.json:3]: We would like to keep versioning here

File: test-app/app/index.html [hunk 4]
```diff
@@ -2,23 +2,22 @@
 <html>
   <head>
     <meta charset="utf-8">
-    <meta http-equiv="X-UA-Compatible" content="IE=edge">
-    <title>amount-input</title>
+    <title>TestApp</title>
```
Comment (dbendaou) [inline test-app/app/index.html:5]:
```suggestion
    <title>ember-amount-input</title>
```

File: test-app/app/index.html [hunk 5]
```diff
@@ -2,23 +2,22 @@
...
+    <link integrity="" rel="stylesheet" href="{{rootURL}}assets/test-app.css">
 
     {{content-for "head-footer"}}
   </head>
   <body>
     {{content-for "body"}}
 
     <script src="{{rootURL}}assets/vendor.js"></script>
-    <script src="{{rootURL}}assets/dummy.js"></script>
+    <script src="{{rootURL}}assets/test-app.js"></script>
```
Comment (dbendaou) [inline test-app/app/index.html:20]: Same

File: test-app/app/router.js [hunk 6]
```diff
@@ -0,0 +1,9 @@
+import EmberRouter from '@ember/routing/router';
+import config from 'test-app/config/environment';
+
+export default class Router extends EmberRouter {
+  location = config.locationType;
+  rootURL = config.rootURL;
+}
+
+Router.map(function () {});
```
Comment (dbendaou) [inline test-app/app/router.js:9]: Need to backport the content of the old dummy app route file 


```
import AddonDocsRouter, { docsRoute } from 'ember-cli-addon-docs/router'
import config from './config/environment'

class Router extends AddonDocsRouter {
  location = config.locationType
  rootURL = config.rootURL
}

Router.map(function() {
  docsRoute(this, function() {
    this.route('installation')

    this.route('components', function() {
      this.route('amount-input')
    })
  })
})

export default Router
```

File: test-app/config/ember-cli-update.json [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+      "blueprints": [
+        {
+          "name": "app",
+          "outputRepo": "https://github.com/ember-cli/ember-new-output",
+          "codemodsSource": "ember-app-codemods-manifest@1",
+          "isBaseBlueprint": true,
+          "options": [
+            "--no-welcome",
+            "--yarn",
+            "--ci-provider=travis"
```
Comment (dbendaou) [inline test-app/config/ember-cli-update.json:16]: We can remove travis

File: test-app/ember-cli-build.js [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+  let app = new EmberApp(defaults, {
+    autoImport: {
+      watchDependencies: ['ember-amount-input'],
+    },
+    'ember-cli-addon-docs': {
+      documentingAddonAt: '../ember-amount-input',
+    },
+  });
+
+  const { maybeEmbroider } = require('@embroider/test-setup');
```
Comment (dbendaou) [inline test-app/ember-cli-build.js:15]: We're also introducing embroider, should we introduce it in a third MR first?

File: test-app/package.json [hunk 9]
```diff
@@ -0,0 +1,92 @@
+{
+  "name": "test-app",
+  "version": "0.0.0",
```
Comment (dbendaou) [inline test-app/package.json:3]: Test app should keep the same versioning for consistency

File: tests/dummy/app/templates/docs.hbs [hunk 10]
```diff
@@ -1,19 +0,0 @@
-<DocsViewer as |viewer|>
```
Comment (dbendaou) [inline tests/dummy/app/templates/docs.hbs:1]: Needs migration

File: tests/integration/components/amount-input/component-test.js
Comment (dbendaou) [inline tests/integration/components/amount-input/component-test.js:1]: Needs migration

File: package.json (hunk 3 above)
Comment (anas7asia) [reply package.json:3]: Should the version be changed only in the addon? Or in both the addon and root package.json?

Comment (dbendaou) [reply package.json:3]: Both, it's should be handle by the release process but we would like to keep versionning in both places

**Comment by anas7asia** [comment]
Closing because this PR has been already merged: https://github.com/qonto/ember-amount-input/pull/431

**Pull request closed without merging**